### PR TITLE
New entry mutable variables

### LINKED ISSUE #5473: [Term Entry] C++ Variables Mutable Variables
### Reason/inspiration (optional)

We would like a new entry on the `Mutable Variables` term under C++. The entry should go in a new file under `docs/content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md`.

The entry should include:  
  
- An introduction to the concept  
- A `Syntax` section that provides the syntax for the concept  
- An `Example` section that provides an example demonstrating the concept in use  
- A `Codebyte Example` section that provides a codebyte example demonstrating the concept in use  
  
Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by braanj**
### Description

Add new concept entry term for cpp mutable variables.

### Issue Solved

Closes #5473 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

**Comment by braanj** [comment]
Hey @codecademy-docs please review

**Comment by braanj** [comment]
Hey @Sriparno08 please review

**Comment by Sriparno08** [comment]
Hey @braanj, I'll review this by the end of the week.

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+Description: 'In C++, the `mutable` keyword in C++ allows certain class variables to be modified even within const functions, useful for managing internal state flexibly.'
+Subjects:
+  - 'Computer Science'
+  - 'Game Development'
+Tags:
+  - 'Variables'
+CatalogContent:
+  - 'learn-c-plus-plus'
+  - 'paths/computer-science'
+---
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:12]: Modifications:

- Let's write the title in title case
- Let's change the description a bit
- Let's add some more tags

```suggestion
---
Title: 'Mutable Variables'
Description: 'Mutable variables are variables that can be modified even within constant functions.'
Subjects:
  - 'Computer Science'
  - 'Game Development'
Tags:
  - 'Functions'
  - 'Values'
  - 'Variable Types'
  - 'Variables'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+In C++, the `mutable` keyword in C++ allows certain class variables to be modified even within const functions, useful for managing internal state flexibly.
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:14]: Let's make some minor changes:

```suggestion
In C++, mutable variables are variables that can be modified even within constant functions, useful for managing internal state flexibly.
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Syntax
+
+To declare a variable as `mutable`, simply place the `mutable` keyword before the variable type:
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:18]: Let's make some minor changes:

```suggestion
To declare a variable as mutable, the `mutable` keyword needs to be placed before the variable type:
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+```pseudo
+mutable type name;
+```
+
+- **type** - The type of the variable (e.g., `int`, `char`).
+- **name** - The name of the variable.
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:25]: Let's make some minor changes:

```suggestion
- `type`: The type of the variable (e.g., `int`, `char`).
- `name`: The name of the variable.
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+## Example
+
+In the example below, `accessCount` is marked `mutable`, allowing it to be modified within the const `displayData` function:
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:29]: Let's make some minor changes:

```suggestion
In the example below, `accessCount` is marked `mutable`, allowing it to be modified within the constant `displayData()` function:
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    mutable int accessCount; // Can be modified in const methods
+};
+
+int main() {
+    Data d("Sample");
+    d.displayData();
+    d.displayData();
+    return 0;
+}
+```
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:55]: Let's change the indentation to 2 spaces and make some minor changes:

````suggestion
```cpp
#include <iostream>
#include <string>

class Data {
public:
  Data(std::string value) : data(value), accessCount(0) {}
  
  void displayData() const {
    ++accessCount; // Modification allowed due to 'mutable'
    std::cout << "Data: " << data << ", Access count: " << accessCount << std::endl;
  }

private:
  std::string data;
  mutable int accessCount; // Can be modified in constant methods
};

int main() {
  Data d("Sample");
  d.displayData();
  d.displayData();
  return 0;
}
```
````

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+
+```shell
+Data: Sample, Access count: 1
+Data: Sample, Access count: 2
+```
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:60]: Let's add a sentence to introduce the output:

````suggestion
The above code produces the following output:

```shell
Data: Sample, Access count: 1
Data: Sample, Access count: 2
```
````

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+
+Here, even though `displayData` is a const member function, `accessCount` can be incremented due to its `mutable` declaration.
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:62]: Let's make some minor changes:

```suggestion
Here, even though `displayData()` is a constant member function, `accessCount` can be incremented due to its `mutable` declaration.
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+
+## Codebyte Example
+
+Use this example to experiment with mutable variables in C++. Enjoy coding!
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:66]: Let's change the sentence:

```suggestion
The following codebyte example demonstrates the usage of mutable variables:
```

File: content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+};
+
+int main() {
+    Counter counter;
+    counter.increment();
+    counter.increment();
+    std::cout << "Count: " << counter.getCount() << std::endl;
+    return 0;
+}
+```
```
Comment (Sriparno08) [inline content/cpp/concepts/variables/terms/mutable-variables/mutable-variables.md:94]: Let's change the indentation to 2 spaces and make some minor changes:

````suggestion
```codebyte/cpp
#include <iostream>

class Counter {
public:
  Counter() : count(0) {}
  
  void increment() const {
    ++count;  // 'count' is mutable, so this modification is allowed in this constant method
  }
  
  int getCount() const {
    return count;
  }

private:
  mutable int count;
};

int main() {
  Counter counter;
  counter.increment();
  counter.increment();
  std::cout << "Count: " << counter.getCount() << std::endl;
  return 0;
}
```
````

**Review by Sriparno08 — CHANGES_REQUESTED** [review]
Hey @braanj, thank you for contributing to Codecademy Docs!

I've listed all the modifications that you need to make in your entry below:

**Review by Sriparno08 — APPROVED** [review]
Hey @braanj, this entry looks good for a second round of review!

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @braanj 😄 

The entry looks good for a second round of review! 🚀

**Pull request merged**